ptf/saimpls.py

```diff
@@ -0,0 +1,3944 @@
+        
+        mpls_tag['ttl'] = 55
+        recv_mpls_tag_list = [mpls_tag]
+        recv_mpls_pkt = simple_mpls_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            mpls_tags=recv_mpls_tag_list,
+            inner_frame=send_pkt['IPv6'])
+
+        mpls_tag['ttl'] = 60
+        recv_mpls_tag_list = [mpls_tag]
+        recv_mpls_pkt_2 = simple_mpls_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            mpls_tags=recv_mpls_tag_list,
+            inner_frame=send_pkt['IPv6'])
+
+        try:
+            print("Send MPLS tag packet with label 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IPv6'].hlim = 59
+            send_packet(self, self.dev_port10, mpls_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Send MPLS tag packet with labels - 2111, 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IPv6'].hlim = 57
+            send_packet(self, self.dev_port10, mpls_tag_2_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Send MPLS tag packet with labels - 2222, 2111, 2000 - "
+                  "PHP and forward IP packet")
+            recv_pkt['IPv6'].hlim = 55
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222,
+                pop_ttl_mode=SAI_INSEG_ENTRY_POP_TTL_MODE_PIPE)
+
+            recv_pkt['IPv6'].hlim = 64
+            print("Send 3 label MPLS packet after changing ttl_mode to pipe - "
+                  "PHP and forward IP packet")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_pkt, self.dev_port13)
+
+            print("Set MPLS object label 2222 to pop only 2 labels")
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=2)
+            print("Send MPLS tag packet with labels - 2222, 2111, 2000 - "
+                  "Pop 2 labels and forward MPLS with 2000")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_mpls_pkt_2, self.dev_port13)
+
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222,
+                pop_ttl_mode=SAI_INSEG_ENTRY_POP_TTL_MODE_UNIFORM)
+
+            print("Send 3 label MPLS packet after changing ttl_mode to "
+                  "uniform - Pop 2 labels and forward MPLS with 2000")
+            send_packet(self, self.dev_port10, mpls_tag_3_pkt)
+            verify_packet(self, recv_mpls_pkt, self.dev_port13)
+
+        finally:
+            sai_thrift_set_inseg_entry_attribute(
+                self.client, self.inseg_entry_2222, num_of_pop=3)
+
+    def mplsTransitPushTtlTest(self):
```

Comment:
I no longer believe that the TTL values in this test are correct from a standards point of view.
This test should be deleted for now, can be restored once we put in a fix.

I opened SWI-4749 to track this.